models/base.py

```diff
@@ -196,8 +202,16 @@ def make_layer_dict(self, *args, **kwargs) -> LayerDictRaw:
     assert name_ctx.maker is self
     name_ctx.is_subnet_ctx = True
     res = self._subnet_func(*args, **kwargs)
-    Copy()(res, name="output")
-    return self._make_layer_dict_from_subnet_ctx(name_ctx)
+    # check if we receive more than one layer as return of a subnet
+    if isinstance(res, (Layer, type(List[Layer]))):
+      Copy()(res, name="output")
+      return self._make_layer_dict_from_subnet_ctx(name_ctx)
+    else:
+      # we return more than one layer (thus also working on other layers of the subnet, that are not output
+      # by convention: first layer is the output layer
+      assert isinstance(res, Tuple)
+      Copy()(res[0], name="output")
+      return (self._make_layer_dict_from_subnet_ctx(name_ctx), *res[1:])
```

Comment:
Remove the brackets.
Also, return type is wrong.
```suggestion
      return self._make_layer_dict_from_subnet_ctx(name_ctx), nest.pack_sequence_as(...)
```
Also, you need to fix the return type annotation of the function (`make_layer_dict`).

Reply:
I will fix the annotation tomorrow but I agree on the suggested changes, so I will commit them and add the typing tomorrow.

Reply:
Now that I think about it, I am not quite sure again. `nest.pack_sequence_as` from my understanding restores the structure of the flattened sequence but doesn't that mean that in this case we return the output layer twice, since it is the result of a Copy layer of the first entry of `res`. Is this intended or am I not getting something here? My current understand would want to do something like `nest.pack_sequence_as(res[1:], res_flat[1:])`.

Reply:
No, we return the layer dict + the nested structure (of layer refs etc).

Reply:
I think I understand it now, cf. other comments.